lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -1273,6 +1274,56 @@ def test_basic(self, mock_emit):
             }
         )
 
+    @ddt.data(
+        *itertools.product(
+            [
+                FORUM_ROLE_ADMINISTRATOR,
+                FORUM_ROLE_MODERATOR,
+                FORUM_ROLE_COMMUNITY_TA,
+                FORUM_ROLE_STUDENT,
+            ],
+            [True, False],
+            [True, False],
+            ["no_group", "group_not_set", "group_set"],
+        )
+    )
+    @ddt.unpack
+    def test_group_id(self, role_name, course_is_cohorted, topic_is_cohorted, data_group_state):
```

Comment:
This test is looking really funky to me for some reason. In other group related tests the group_states are ["no_group", "match_group", "different_group"] and behave differently such as when the no_group is set, the group_id=none whereas in this function, group_not_set=none. Still trying to wrap my head around this test and will address in the next fixup.